tests/D2L.CodeStyle.Analyzers.Test/Specs/OnlyVisibleToAnalyzer.cs

```diff
@@ -234,6 +234,28 @@ public static class AlwaysVisibleWithinSameContainer {
 			Fly();
 		}
 	}
+
+	public interface IRestrictedInterface {
+
+		[OnlyVisibleToType( typeof( string ) )]
+		void RestrictedMethod();
+	}
+
+	public static class RestrictedMembersOnSelf : IRestrictedInterface {
+
+		[OnlyVisibleToType( typeof( string ) )]
+		public static void RestrictedMethod() {}
+
+		public static void Caller() {
+
+			// Ok when called on this
+			this.RestrictedMethod();
+
+			// Violation when called via interface, since not in set
+			IRestrictedInterface @interface = this;
+			/* MemberNotVisibleToCaller(RestrictedMethod) */ @interface.RestrictedMethod() /**/;
+		}
+	}

```

Comment:
```suggestion
	public class RestrictedMembersOnSelf : IRestrictedInterface {

		[OnlyVisibleToType( typeof( string ) )]
		public void RestrictedMethod() {}

		public void Caller() {

			// Ok when called on this
			this.RestrictedMethod();

			// Violation when called via interface, since not in set
			IRestrictedInterface @interface = this;
			/* MemberNotVisibleToCaller(RestrictedMethod) */ @interface.RestrictedMethod() /**/;
		}
	}
```